src/test/ui/lint/lint-min-warn-cannot-allow.rs

```diff
@@ -0,0 +1,17 @@
+#![deny(warnings)]
+//~^ NOTE lint level defined here
+
+#![allow(illegal_floating_point_literal_pattern)]
+
+fn _0() {
+    if let 0.0 = 0.0 {}
+    //~^ ERROR floating-point types cannot be used in patterns
```

Comment:
I ... am not sure I would expect this behavior either, as a user.

If the user writes:

```rust
#![deny(warnings)]
#![warn(illegal_floating_point_literal_pattern)]
```

what happens then? Does that let the warning through in that case, avoiding the error?

----

My intuition here is that if someone requests `Allow` (as an explicit exception to a standing `#![deny(warnings)]`, and the min-level rejects that setting, the user probably would prefer the level be downgraded to `Warn` rather than `Deny` here.

Reply:
Ah!

It turns out that the behavior that I expected the user to desire cannot even be expressed today!

----

In particular, I had thought we had some way to say "deny all warnings, but let me carve out subsets of the warnigns for which I still want to see warnings (but *not* errors)."

But it turns out our lint system today **cannot express this**

Example ([play](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=33fe8e7e01fb7148f6d6f1bddc9949c4)):

```rust
#![deny(warnings)]

mod m {
    #![warn(non_camel_case_types)]
    pub struct foo;
}
```

causes an error today.

----

So, I withdraw my objection to the behavior shown in this test. It is consistent with our current behavior (a current behavior that I may or may not think unfortunate; I guess I am of two minds about it, because I can understand wanting instances of `#![warn(...)]` in a code base to be *included* in whatever setting is in place for `#![deny(warnings)]`...)